.github/workflows/pr-ci.yaml

```diff
@@ -0,0 +1,34 @@
+name: CI
+on: [pull_request]
+permissions:
+  contents: read
+jobs:
+  lint-code:
+    name: golangci-lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Prepare GO env
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.17'
+          cache: false
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: v1.49.0
+
+  build-code:
+    name: sample-build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Prepare GO env
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.17'
```

Comment:
nit: Shouldn't we use a newer Go version?

Reply:
yes, let update it to 1.19 same in go mod. 
https://github.com/Mellanox/mlxdevm-go/blob/main/go.mod#L3

Reply:
Done